### PR TITLE
Integrate tokens endpoints - Closes #4404

### LINKED ISSUE #4404: Integrate tokens endpoints
### Description

- Replace mock endpoints with actual endpoints to fetch
   - Supported tokens 
   - Token and its balances
  - Get token meta data from `/api/v3/blockchain/apps/meta/tokens`
  - Get token balances (available, locked) from `/api/v3/tokens` or RPC `token_getBalances`
  - Get nonce and multisignature information from `auth_getAuthAccount`
  - Get List of voters from `​​dpos_getVoter`
  - Get delegate information from `dpos_getDelegate`
  - Get supported tokens from selected `to application` to filter the un-supported tokens to send to receiving chain
 

### Acceptance Criteria

- Users should be able to select the supported tokens and send them to users
- Users should be able to select the supported tokens and request them from users
- Add e2e tests for
   - Send LSK tokens within application
   - Send LSK tokens across application
   - Send native tokens within application
   - Send native tokens across application
   - Receive LSK tokens within application
   - Receive LSK tokens across application
   - Receive native tokens within application
   - Receive native tokens across application

### PR DESCRIPTION
**Pull request by soroushm**
### What was the problem?

This PR resolves #4404

### How was it solved?
- use react-query hooks

### How was it tested?
by unit test

### CODE REVIEW

File: src/modules/token/fungible/hooks/useTokensBalanceSupportedFrom.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useMemo, useRef } from 'react';
+import { useTokensBalance, useTokensSupported } from '@token/fungible/hooks/queries';
+import {Client} from 'src/utils/api/client';
+// eslint-disable-next-line import/prefer-default-export
+export const useTokensBalanceSupportedFrom = (application) => {
```
Comment (clemente-xyz) [inline src/modules/token/fungible/hooks/useTokensBalanceSupportedFrom.js]: why dont we name this query just `useApplicationSupportedTokens`?

Comment (clemente-xyz) [reply src/modules/token/fungible/hooks/useTokensBalanceSupportedFrom.js]: the current name is a bit confusing

File: src/modules/token/fungible/hooks/queries/useTokensSupported.js [hunk 2]
```diff
@@ -18,17 +18,19 @@ import { useCustomInfiniteQuery } from 'src/modules/common/hooks';
  * @returns the query object
  */
 // eslint-disable-next-line import/prefer-default-export
-export const useTokensSupported = ({ config: customConfig = {}, options } = {}) => {
+export const useTokensSupported = ({ config: customConfig = {}, options } = {}, client) => {
```
Comment (clemente-xyz) [inline src/modules/token/fungible/hooks/queries/useTokensSupported.js]: i think `useSupportedTokens` is a more descriptive name or not?

Comment (clemente-xyz) [reply src/modules/token/fungible/hooks/queries/useTokensSupported.js]: shouldn't the `client` arg go inside the main object arg? You're invoquing `useTokensSupported` and passing `client` as field of an object arg.

File: src/utils/api/client.js [hunk 3]
```diff
@@ -7,15 +7,28 @@ export class Client {
...
+          return reject(response);
+        }
+        return resolve(response);
+      });
+    });
 
-  http = null
+  rest = (config) => this.http?.request({ ...this.http.defaults, ...config });
 
   create({ rpc, rest }) {
```
Comment (clemente-xyz) [inline src/utils/api/client.js]: idk why, but you have this deprecated. Should be `{ws, http}` instead (according to the new service API schema)

File: src/modules/common/components/ApplicationField/ApplicationField.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import MenuSelect, {
+  MenuItem
+} from 'src/modules/wallet/components/MenuSelect';
+import chainLogo from '@setup/react/assets/images/LISK.png';
+// import Skeleton from 'src/modules/common/components/skeleton';
```
Comment (clemente-xyz) [inline src/modules/common/components/ApplicationField/ApplicationField.js]: please remove this comment if no needed

File: src/modules/common/hooks/queries/useCcm.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ import { useCustomInfiniteQuery } from '../useCustomInfiniteQuery';
  * @returns the query object
  */
 // eslint-disable-next-line import/prefer-default-export
-export const useCcm = ({ config: customConfig = {}, options } = { }) => {
+export const useCcm = ({ config: customConfig = {}, options } = { }) => { // deprecated by service
```
Comment (clemente-xyz) [inline src/modules/common/hooks/queries/useCcm.js]: can you deprecate it via jsdoc? (add `@deprecated` decorator)

File: src/modules/token/fungible/components/SendForm/form.test.js [hunk 6]
```diff
@@ -24,20 +24,24 @@ jest.mock('@blockchainApplication/manage/hooks/useApplicationManagement');
 jest.mock('@blockchainApplication/manage/hooks/useCurrentApplication');
 jest.mock('@account/hooks/useCurrentAccount');
 jest.mock('@token/fungible/hooks/queries');
+// jest.mock('../../hooks/useMessageField');
+// useMessageField.mockImplementation(jest.requireActual('../../hooks').useMessageField); // loop issue
```
Comment (clemente-xyz) [inline src/modules/token/fungible/components/SendForm/form.test.js]: can you remove this comments?

**Review by clemente-xyz — COMMENTED** [review]
left more comments

**Review by clemente-xyz — DISMISSED** [review]

**Review by ManuGowda — APPROVED** [review]
Approving this PR as of now, however @soroushm and @Balanced02 will test the token endpoint integration in another PR and ensure to fix or create issues that are not working from this PR changes.

**Pull request merged**